cli/src/commands/show.rs

```diff
@@ -50,6 +50,9 @@ pub(crate) struct ShowArgs {
     template: Option<String>,
     #[command(flatten)]
     format: DiffFormatArgs,
+    /// Do not show the patch.
+    #[arg(long, short = 'P')]
```

Comment:
nit: 

* add `conflicts_with = "DiffFormatArgs"` to reject `--no-patch --git`, etc.
* remove period from summary line of help text
* remove short flag for now? I'm not sure if we need short flag.

Reply:
Out of curiosity, what's your use case? (I assumed `show --no-patch -T` could be useful in scripting, but wouldn't be used much as interactive command.)

I don't have strong feeling either, but `jj show -P` doesn't sound like an option to suppress patch.